### PR TITLE
Bring zipkin-finagle up to date

### PR DESCRIPTION
**Pull request by reta**
Bring zipkin-finagle up to date

### CODE REVIEW

**Comment by reta** [comment]
@codefromthecrypt would really appreciate if you could find a time to look at it: this is first step to bring the project back to shape, the other things (Docker, Github Actions, SPDX licenses, ... ) to follow shortly, thanks!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -16,15 +16,18 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-20.04  # newest available distribution, aka focal
+    runs-on: ubuntu-latest
```
Comment (codefromthecrypt) [inline .github/workflows/test.yml:19]: nit I think this undoes the value of knowing which dist it is..

File: core/pom.xml [hunk 2]
```diff
@@ -19,12 +19,12 @@
 
   <parent>
     <groupId>io.zipkin.finagle2</groupId>
-    <artifactId>zipkin-finagle-parent_2.12</artifactId>
-    <version>22.4.0</version>
+    <artifactId>zipkin-finagle-parent_2.13</artifactId>
```
Comment (codefromthecrypt) [inline core/pom.xml:22]: hmm I thought in our build we did some tricks to dual publish to 12 and 13 via replacement?

**Review by codefromthecrypt — APPROVED** [review]
looks good, but maybe revert the move to scala 2.13 unless finagle is only 2.13 now.. in any case look at build-bin/deploy for what I mean

**Comment by reta** [comment]
> looks good, but maybe revert the move to scala 2.13 unless finagle is only 2.13 now.. in any case look at build-bin/deploy for what I mean

It is both, you are very right, I will take care of that, thanks!

File: .github/workflows/test.yml (hunk 1 above)
Comment (reta) [reply .github/workflows/test.yml:19]: Noted!

**Comment by codefromthecrypt** [comment]
another thing I think we should figure out how to turn off scala-steward and close all the unaddressed spam PRs..

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -16,15 +16,18 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-20.04  # newest available distribution, aka focal
+    runs-on: ubuntu-24.04
```
Comment (codefromthecrypt) [inline .github/workflows/test.yml:19]:
```suggestion
    runs-on: ubuntu-22.04. # Hard-coding an LTS means maintenance, but only once each 2 years!
```

Comment (codefromthecrypt) [reply .github/workflows/test.yml:19]: I think I left out the why here ;) (about most recent LTS)

**Review by codefromthecrypt — APPROVED** [review]

**Comment by reta** [comment]
> another thing I think we should figure out how to turn off scala-steward and close all the unaddressed spam PRs..

Sure, I think once we merge this one, many will be auto closed

**Comment by codefromthecrypt** [comment]
Thanks @reta!

**Pull request merged**